ui/ducks/bridge/actions.ts

```diff
@@ -86,3 +118,330 @@ export const updateQuoteRequestParams = (params: Partial<QuoteRequest>) => {
     );
   };
 };
+
+export const getBridgeERC20Allowance = async (
+  contractAddress: string,
+  chainId: Hex,
+): Promise<string> => {
+  return await submitRequestToBackground(
+    BridgeBackgroundAction.GET_BRIDGE_ERC20_ALLOWANCE,
+    [contractAddress, chainId],
+  );
+};
+
+export const submitBridgeTransaction = (
```

Comment:
Rather than including lots of logic in this monolithic file, would it be more extensible and readable to break it down into multiple React hooks you can use from within the bridge React components?

This actions file can still exist, but just to provide thin wrappers around the background API methods?